modules/web/services/darkhttpd/default.nix

```diff
@@ -0,0 +1,206 @@
+# page that informs them what to do next. TODO: Not active yet.
+
+# Configuration of this module aims to be extremely simple, which
+# makes it easy to get started and allows this to be easily copied.
+# This means that not everyone trying out this webservice is
+# likely not have had a thorough read of the documentation. Also, if
+# you use the default, you may not know the contrived location where
+# you need to put your data.
+
+let
+    serviceName = "darkhttpd";
+
+    placeholderInfo = pkgs.writeText "placeholder.html" ''
+     <h1>Congratulations. Its working!</h1>
+     <p>You have just created a webservice to serve up your static
+        web content. You should copy your content here:</p>
+      <p style="text-align: center">${config.staticDir}</p>
+      <p>The documentation for this can be found at https://nixcloud.io</p>
+  '';
+in
+
+{
+  options = {
+    staticDir = mkOption {
+    type = types.path;
+    default = config.stateDir;
+      description = ''
+        Serve static content from the directory specified. By default it is a
+        dedicated (new) subdirectory under the state directory unique to
+        your service instance. Obviously, you will have to get your content into
+        that directory.
+      '';
+    };
+    cloneDir = mkOption {
+    type = types.bool;
+    default = false;
+      description = ''
+        Serve the content from a cloned copy of the directory indicated by the user.
+        The copy is created in a unique folder under /var/run/nixcloud/webservices
+        aka the 'state directory'. Especially useful for tiny websites that want to
+        keep the original content somewhere safe.
+      '';
+    };
+    serviceOwner = mkOption {
+    type = types.string;
+    default = builtins.replaceStrings ["."] ["_"] config.proxyOptions.domain;
+      description = ''
+	The group that is to be given ownership across a set of
+	services to allow access.  The default option is a derivation
+	of the domain name under which the service will be run -- e.g.
+	example_com (for some unknown reason dots cause problems in many
+	cases). However the user may use any valid group name.
+      '';
+    };
+    directoryListing = mkOption {
+    default = false;
+      description = ''
+	If a folder you are serving does not contain a file named
+	index.html, setting the flag directoryListing will instead
+	list the files present in that folder. Note that this may in
+	some cases expose undesired details from your system, and is
+	therefore turned off by default.
+      '';
+    };
+    log = mkOption {
+    default = false;
+      description = ''
+        Whether to log any data about requests to the webserver,
+	or not. If you are not going to use it, because you use
+	some other mechanism (like Piwik) you might as well keep it
+	turned off to save yourself disk wear and enhance user privacy.
+      '';
+    };
+    logFile = mkOption {
+    default = "/var/log/${serviceName}";
```

Comment:
This is IMO a bad idea, because if you have multiple instances, they both use the same location.